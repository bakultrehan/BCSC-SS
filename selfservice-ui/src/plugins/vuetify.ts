import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
