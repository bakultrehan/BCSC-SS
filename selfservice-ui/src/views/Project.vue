<template>
  <div class="home">
    <CreateApp :step="step" :id="id" />
  </div>
</template>

<script lang="ts">
// @ is an alias to /src
import { Component, Vue, Prop } from 'vue-property-decorator';
import CreateApp from '@/components/CreateApp/CreateApp.vue';

@Component({
  components: {
    CreateApp
  }
})
export default class CreateAppHome extends Vue {
  @Prop({ default: 'info' })
  public step!: string;
  @Prop({ default: '' })
  public id!: string;
}
</script>
