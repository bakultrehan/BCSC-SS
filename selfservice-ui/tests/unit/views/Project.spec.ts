import { shallowMount, createLocalVue } from '@vue/test-utils';
import Project from '@/views/Project.vue';
import vuetify from 'vuetify';

describe('Project.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(Project, {
      localVue
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
