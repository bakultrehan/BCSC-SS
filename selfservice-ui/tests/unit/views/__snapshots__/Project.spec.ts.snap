// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project.vue renders props when passed 1`] = `
<div
  class="home"
>
  <createapp-stub
    step="projectinfo"
  />
</div>
`;
